app/connectors/httpParsers/CompaniesHouseHttpParser.scala

```diff
@@ -16,43 +16,28 @@
 
 package connectors.httpParsers
 
+import connectors.HttpHelper.CompaniesHouseResponse
+import connectors.{HttpErrorMessages, InvalidCRN, UnexpectedFailure}
 import play.api.Logger
 import play.api.http.Status._
-import play.api.libs.json.Json
 import uk.gov.hmrc.http.{HttpReads, HttpResponse}
 
 object CompaniesHouseHttpParser {
 
-  type CompaniesHouseResponse = Either[ErrorResponse, SuccessResponse]
-
   implicit object CompaniesHouseReads extends HttpReads[CompaniesHouseResponse] {
 
     override def read(method: String, url: String, response: HttpResponse): CompaniesHouseResponse = {
-
       response.status match {
         case OK =>
           Logger.debug("[CompaniesHouseHttpParser][read]: Status OK")
           Logger.debug(s"[CompaniesHouseHttpParser][read]: Json Response: ${response.json}")
-          Right(ValidCRN)
```

Comment:
I'd rather keep this as a Sealed Trait with two responses. Just because it has more meaning then semantically than just a boolean would.